api/machine/machine.proto

```diff
@@ -30,6 +30,7 @@ service MachineService {
       returns (EtcdLeaveClusterResponse);
   rpc EtcdForfeitLeadership(EtcdForfeitLeadershipRequest)
       returns (EtcdForfeitLeadershipResponse);
+  rpc EtcdSnapshot(EtcdSnapshotRequest) returns (stream common.Data);
```

Comment:
I'm not sure we should share request and response messages. I think that's where duplication is worth it because it will allow us to modify messages independently.

Also, this time is good as any other to start adding API method documentation comments :)

Reply:
I thought about that, but there's benefit here to common response type - we have common layer handling `stream common.Data`, so it's better to have it same vs. special